var/spack/repos/builtin/packages/py-yt/package.py

```diff
@@ -17,46 +17,70 @@ class PyYt(PythonPackage):
-    version("3.2.2", sha256="498ed77b3dae8c54929602d4931f3c3e0a3420a9b500cbd870f50b1e0efea8c3")
+    version("4.1.2", sha256="0ae03288b067721baad14c016f253dc791cd444a1f2dd5d804cf91da622a0c76")
+    version("3.6.1", sha256="be454f9d05dcbe0623328b4df43a1bfd1f0925e516be97399710452931a19bb0")
+    version("3.6.0", sha256="effb00536f19fd2bdc18f67dacd5550b82066a6adce5b928f27a01d7505109ec")
+    version("3.5.1", sha256="c8ef8eceb934dc189d63dc336109fad3002140a9a32b19f38d1812d5d5a30d71")
+    version("3.5.0", sha256="ee4bf8349f02ce21f571654c26d85c1f69d9678fc8f5c7cfe5d1686c7ed2e3ca")
+    version("3.4.1", sha256="a4cfc47fe21683e7a3b680c05fe2a25fb774ffda6e3939a35755e5bf64065895")
+    version("3.4.0", sha256="de52057d1677473a83961d8a1119a9beae3121ec69a4a5469c65348a75096d4c")
+    version("3.3.5", sha256="4d5c751b81b0daf6dcaff6ec0faefd97138c008019b52c043ab93403d71cedf6")
+    version("3.3.4", sha256="64c109ba4baf5afc0e1bc276ed2e3de13f1c5ce85c6d8b4c60e9a47c54bf3bcb")
+    version("3.3.3", sha256="edf6453927eee311d4b51afacb52cd5504b2b57cc6d3d92dab9c6bfaf6d883df")
+    version("3.3.2", sha256="a18e4cf498349804c64eec6509ec4d3a6beaa34ea63366543290c35774337f0e")
+    version("3.3.1", sha256="01e3b3398d43b8eab698d55ba37ef3d701ea026b899c0940a1ee34b215e25a31")
+    version("3.3.0", sha256="537c67e85c8f5cc5530a1223a74d27eb7f11c459651903c3092c6a97b450d019")
+    version("3.2.3", sha256="4d6ccf345d9fab965335c9faf8708c7eea79366b81d77f0f302808be3e82c0ed")
+    version("3.2.2", sha256="78866f51e4751534ad60987000f149a8295952b99b37ca249d45e4d11095a5df")
 
     variant("astropy", default=True, description="enable astropy support")
     variant("h5py", default=True, description="enable h5py support")
     variant("scipy", default=True, description="enable scipy support")
     variant("rockstar", default=False, description="enable rockstar support")
 
-    depends_on("py-astropy@4.0.1:", type=("build", "run"), when="+astropy")
-    depends_on("py-cython@0.24:", type=("build", "run"))
-    depends_on("py-h5py@3.1:", type=("build", "run"), when="+h5py")
-    depends_on("py-ipython@1.0:", type=("build", "run"))
+    # Main dependencies:
+    # See https://github.com/yt-project/yt/blob/yt-4.1.2/setup.cfg#L40
+    depends_on("py-cmyt@0.2.2:", type=("build", "run"), when="@4.1.2:")
+    depends_on("py-ipywidgets@8:", type=("build", "run"), when="@4.1.2")
     depends_on("py-matplotlib@1.5.3:", type=("build", "run"))
     depends_on("py-matplotlib@:3.2.2", type=("build", "run"), when="@:3.6.0")
+    depends_on("py-matplotlib@3.1:", type=("build", "run"), when="@4.1.2:")
+    conflicts("^py-matplotlib@3.4.2", when="@4.1.2:")
+    depends_on("py-more-itertools@8.4:", when="@4.1.2:")
     depends_on("py-numpy@1.10.4:", type=("build", "run"))
-    depends_on("py-scipy@1.5.0:", type=("build", "run"), when="+scipy")
+    depends_on("py-numpy@1.14.5:", type=("build", "run"), when="@4.1.2:")
     depends_on("py-setuptools@19.6:", type=("build", "run"))
```

Comment:
Should setuptools be removed here, or restricted to older versions?

Reply:
Actually, it looks like it is used at run-time, so maybe just change the dep from `pyproject.toml` below to be build/run?

Reply:
I'm not totally sure what you mean; the newest commit changes `setuptools` to be available at runtime.

Reply:
You interpreted me correctly.